File: src/main/java/org/dedda/bratwurst/lang/io/terminal/ReadLine.java
package org.dedda.bratwurst.lang.io.terminal;

import org.dedda.bratwurst.lang.BWExpression;
import org.dedda.bratwurst.lang.BWObject;
import org.dedda.bratwurst.lang.BWString;
import org.dedda.bratwurst.lang.scope.Scope;

import java.io.IOException;
import java.io.InputStreamReader;
import java.io.Reader;
import java.util.Scanner;

/**
 * Created by dedda on 1/28/16.
 *
 * @author dedda
 */
public class ReadLine extends BWExpression {

    private String read;

    public ReadLine(int lineNumber) {
        super(lineNumber);
    }

    @Override
    public BWObject getValue() {
        return new BWString(read);
    }

    @Override
    public int getIntValue() {
        return getValue().getIntValue();
    }

    @Override
    public String getValueType() {
        return "string";
    }

    @Override
    public void run(Scope scope) {
        Scanner scanner = new Scanner(System.in);
        read = scanner.nextLine();
    }
}

File: src/main/java/org/dedda/bratwurst/lang/io/terminal/ReadLine.kt
package org.dedda.bratwurst.lang.io.terminal

import org.dedda.bratwurst.lang.BWExpression
import org.dedda.bratwurst.lang.BWObject
import org.dedda.bratwurst.lang.BWString
import org.dedda.bratwurst.lang.scope.Scope
import java.util.*

/**
 * Created by dedda on 1/28/16.
 *
 * @author dedda
 */
class ReadLine(lineNumber: Int) : BWExpression(lineNumber) {
    private var read: String? = null
    override fun getValue(): BWObject {
        return BWString(read)
    }

    override fun getIntValue(): Int {
        return value.intValue
    }

    override fun getValueType(): String {
        return "string"
    }

    override fun run(scope: Scope) {
        val scanner = Scanner(System.`in`)
        read = scanner.nextLine()
    }
}